src/authentication/json-api-authorization-spec.ts

```diff
@@ -254,11 +314,25 @@ describe('authorization for api routes', () => {
         // Act
         const response = await makeRequest(server, '/comments/1', 'DELETE');
         // Assert
-        expect(api.getCommentProjectId).to.have.been.calledOnce;
-        expect(authentication.userHasAccessToProject).to.have.been.calledOnce;
+        expect(api.getComment).to.have.been.calledOnce;
+        expect(authentication.userHasAccessToDeployment).to.have.been.calledOnce;
         expect(api.deleteCommentHandler).to.not.have.been.called;
         expect(response.statusCode).to.eq(401);
       });
+      it('should allow deleting a comment for an open deployment without authentication', async () => {
```

Comment:
I think we should actually make it so that deleting comments is not allowed if you're not authorized.